src/main.rs

```diff
@@ -91,15 +108,48 @@ fn main() -> Result<(), ureq::Error> {
 
             // Iterate over meanings and print them
             for (i, entry) in body.iter().enumerate() {
-                if i >= options.limit {
+                if i >= options.limit && options.limit != 0 {
                     break;
                 }
-
-                if print_item(&query, entry).is_some() && i + 2 <= options.limit {
-                    println!();
+                match print_item(&query, entry, &mut output) {
+                    Some(r) => lines_output += r,
+                    None => continue,
                 }
+
+                output.push('\n');
+                lines_output += 1;
+            }
+            output.pop();
+            if lines_output > 0 {
+                lines_output -= 1;
             }
-            println!();
+
+        }
+
+        if lines_output >= term_size - 1 && term_size != 0 {
+            // Output is a different process that is not a tty (i.e. less), but we want to keep colour
+            env::set_var("CLICOLOR_FORCE", "1");
+
+            match Command::new("less")
+                            .arg("-R")
+                            .stdin(Stdio::piped())
+                            .spawn() {
+                                Ok(mut process) => {
+                                    if let Err(e) = process.stdin.as_ref().unwrap().write_all(output.as_bytes()) {
+                                        panic!("couldn't pipe to less: {}", e);
+                                    }
+
+                                    // We don't care about the return value, only whether wait failed or not
+                                    if process.wait().is_err() {
+                                        panic!("wait() was called on non-existent child process\
+                                         - this should not be possible");
+                                    }
+                                }
+                                // less not found in PATH; print normally
+                                Err(_e) => print!("{}", output)
+                            };
```

Comment:
Personally I would create a separate function for this. Also please don't write such big matches as they're hard to read and the level of indentation increases by a lot. Put the Command::new()...spawn() in a variable and use that as match condition.

Reply:
Yeah sorry, I realised this was janky only after I pushed the commit.